homeassistant/components/qvrpro/__init__.py

```diff
@@ -0,0 +1,89 @@
+"""Support for QVR Pro NVR software by QNAP."""
+
+import logging
+
+from pyqvrpro import Client
+from pyqvrpro.client import AuthenticationError, InsufficientPermissionsError
+import voluptuous as vol
+
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.discovery import load_platform
+
+from .const import CONF_EXCLUDE_CHANNELS, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_HOST): cv.string,
+                vol.Required(CONF_USERNAME): cv.string,
+                vol.Required(CONF_PASSWORD): cv.string,
+                vol.Optional(CONF_EXCLUDE_CHANNELS, default=[]): vol.All(
+                    cv.ensure_list_csv, [cv.positive_int]
+                ),
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+def setup(hass, config):
+    """Set up the QVR Pro component."""
+    user = config[DOMAIN][CONF_USERNAME]
+    password = config[DOMAIN][CONF_PASSWORD]
+    host = config[DOMAIN][CONF_HOST]
+    excluded_channels = config[DOMAIN][CONF_EXCLUDE_CHANNELS]
+
+    try:
+        qvrpro = Client(user, password, host)
+
+        channel_resp = qvrpro.get_channel_list()
+
+    except InsufficientPermissionsError:
+        _LOGGER.error("User must have Surveillance Management permission")
+        return False
+    except AuthenticationError:
+        _LOGGER.error("Authentication failed")
+        return False
+
+    channels = []
+
+    for channel in channel_resp["channels"]:
+        if channel["channel_index"] + 1 in excluded_channels:
+            continue
+
+        channels.append(QVRChannel(**channel))
+
+    hass.data[DOMAIN] = {"channels": channels, "client": qvrpro}
+
+    load_platform(hass, "camera", DOMAIN, {}, config)
+
+    # Register services
+    def handle_start_record(call):
+        guid = call.data.get("guid")
+        qvrpro.start_recording(guid)
```

Comment:
What happens if the GUID does not exist?

Reply:
Nothing happens. The API still returns a 200 response with a flag that indicates the call was not successful.  

![image](https://user-images.githubusercontent.com/453840/74076882-fd561300-49e0-11ea-9721-d81b80499c88.png)

Reply:
Okay good to know.